lib/rules/selector-no-vendor-prefix/index.js

```diff
@@ -49,25 +49,29 @@ const rule = (primary, secondaryOptions, context) => {
 
 			parseSelector(selector, result, ruleNode, (selectorTree) => {
 				selectorTree.walkPseudos((pseudoNode) => {
-					if (isAutoprefixable.selector(pseudoNode.value)) {
-						if (optionsMatches(secondaryOptions, 'ignoreSelectors', pseudoNode.value)) {
-							return;
-						}
+					const { value } = pseudoNode;
 
-						if (context.fix) {
-							ruleNode.selector = isAutoprefixable.unprefix(ruleNode.selector);
+					if (!isAutoprefixable.selector(value)) {
+						return;
+					}
```

Comment:
[note] Using an early return for `isAutoprefixable.selector(value)` reduces a nest level.

Note: [Hiding whitespaces](https://github.com/stylelint/stylelint/pull/6261/files?w=1) makes it easier to review the diff.